avm/res/hybrid-compute/machine/main.bicep

```
@@ -0,0 +1,256 @@
+        assessmentMode: patchAssessmentMode
+      }
+    : null
+}
+
+var windowsConfiguration = {
+  patchSettings: (patchMode =~ 'AutomaticByPlatform' || patchMode =~ 'AutomaticByOS' || patchMode =~ 'Manual')
+    ? {
+        patchMode: patchMode
+        assessmentMode: patchAssessmentMode
+        // enableHotpatching: enableHotpatching // support added in 2024-05-20-preview
+      }
+    : null
+}
+
+var builtInRoleNames = {
+  Contributor: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b24988ac-6180-42a0-ab88-20f7382dd24c')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'f58310d9-a9f6-439a-9e8d-f62e7b41a168'
+  )
+  'User Access Administrator': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9'
+  )
+  'Arc machine Administrator Login': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '1c0163c0-47e6-4577-8991-ea5c82e286e4'
+  )
+  'Arc machine Contributor': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    '9980e02c-c2be-4d73-94e8-173b1dc7cf3c'
+  )
+  'Arc machine User Login': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'fb879df8-f326-4884-b1cf-06f3ad86be52'
+  )
+  'Windows Admin Center Administrator Login': subscriptionResourceId(
+    'Microsoft.Authorization/roleDefinitions',
+    'a6333a3e-0164-44c3-b281-7a577aff287f'
+  )
+}
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.hybridcompute-machine.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource machine 'Microsoft.HybridCompute/machines@2024-03-31-preview' = {
+  name: name
+  location: location
+  identity: {
+    type: 'SystemAssigned'
+  }
+  tags: tags
+  kind: kind
+  properties: {
+    osProfile: {
+      windowsConfiguration: osType == 'Windows' ? windowsConfiguration : null
+      linuxConfiguration: osType == 'Linux' ? linuxConfiguration : null
+    }
+    parentClusterResourceId: parentClusterResourceId
+    vmId: vmId
+    clientPublicKey: clientPublicKey
+    privateLinkScopeResourceId: empty(privateLinkScopeResourceId) ? null : privateLinkScopeResourceId
```

```suggestion
    privateLinkScopeResourceId: !empty(privateLinkScopeResourceId) ? privateLinkScopeResourceId : null 
```

Having the actual value first is usually 'easier' to read :)